pkg/apis/triggers/v1alpha1/event_listener_types.go

```diff
@@ -60,8 +63,72 @@ type EventListenerTemplate struct {
 	APIVersion string `json:"apiversion,omitempty"`
 }
 
-// EventListenerStatus defines the observed state of EventListener
-type EventListenerStatus struct{}
+// The conditions that are internally resolved by the EventListener reconciler
+const (
+	// ServiceExists is the ConditionType set on the EventListener, which specifies Service existence.
+	ServiceExists apis.ConditionType = "Service"
+	// DeploymentExists is the ConditionType set on the EventListener, which specifies Deployment existence.
+	DeploymentExists apis.ConditionType = "Deployment"
+)
+
+var eventListenerCondSet = apis.NewLivingConditionSet(ServiceExists, DeploymentExists)
+
+// SetExistsConditions simplifies setting the exists conditions on the EventListenerStatus
+func (els *EventListenerStatus) SetExistsCondition(cond apis.ConditionType, err error) {
+	if err != nil {
+		els.SetCondition(&apis.Condition{
+			Type:    cond,
+			Status:  corev1.ConditionFalse,
+			Message: err.Error(),
+		})
+	} else {
+		els.SetCondition(&apis.Condition{
+			Type:    cond,
+			Status:  corev1.ConditionTrue,
+			Message: fmt.Sprintf("%s exists", cond),
+		})
+	}
+}
+
+// InitializeConditions will set all conditions in eventListenerCondSet to unknown for the EventListener.
+// This does not use the InitializeCondition() provided by the conditionsImpl to avoid setting the happy condition.
+// This is a local change and needs to be persisted to the K8s API elsewhere.
+func (els *EventListenerStatus) InitializeConditions() {
+	for _, condition := range []apis.ConditionType{
+		ServiceExists,
+		DeploymentExists,
+	} {
+		els.SetCondition(&apis.Condition{
+			Type:   condition,
+			Status: corev1.ConditionFalse,
```

Comment:
> InitializeConditions will set all conditions in eventListenerCondSet to unknown for the EventListener.

I think this should be `core.v1ConditionUnknown` like it says in the comment. Is the comment or the code wrong? 😁

Reply:
I will update the docstring to false